src/libraries/System.Runtime.InteropServices/gen/Common/ConvertToSourceGeneratedInteropFixer.cs

```diff
@@ -259,5 +260,45 @@ static SyntaxNode GenerateMarshalAsUnmanagedTypeBoolAttribute(SyntaxGenerator ge
                              generator.DottedName(TypeNames.System_Runtime_InteropServices_UnmanagedType),
                              generator.IdentifierName(unmanagedTypeMemberIdentifier))));
         }
+
+        protected static SyntaxNode AddHResultStructAsErrorMarshalling(SyntaxGenerator generator, IMethodSymbol methodSymbol, SyntaxNode generatedDeclaration)
+        {
+            foreach (IParameterSymbol parameter in methodSymbol.Parameters)
+            {
+                if (parameter.Type is { TypeKind: TypeKind.Struct }
+                    && IsHResultLikeType(methodSymbol.ReturnType)
+                    && !parameter.GetAttributes().Any(attr => attr.AttributeClass?.ToDisplayString() == TypeNames.System_Runtime_InteropServices_MarshalAsAttribute))
+                {
+                    SyntaxNode generatedParameterSyntax = generator.GetParameters(generatedDeclaration)[parameter.Ordinal];
+                    generatedDeclaration = generator.ReplaceNode(generatedDeclaration, generatedParameterSyntax, generator.AddAttributes(generatedParameterSyntax,
+                                    GeneratedMarshalAsUnmanagedTypeErrorAttribute(generator)));
+                }
+            }
```

Comment:
Do we want to be replacing parameters that are structs if the return type is HResult-like? Should this be removed?